docs/diagrams/migration-console-library-source.md

```diff
@@ -0,0 +1,261 @@
+---
+
+excalidraw-plugin: parsed
+tags: [excalidraw]
+
+---
+==⚠  Switch to EXCALIDRAW VIEW in the MORE OPTIONS menu of this document. ⚠== You can decompress Drawing data with the command palette: 'Decompress current Excalidraw file'. For more info check in plugin settings under 'Saving'
+
+
+# Excalidraw Data
+## Text Elements
+CLI ^lbKc6IRp
+
+Web API ^C64hjNOd
+
+Library ^23uvP31d
+
+Source Cluster
+on self-managed EC2 ^t5vxFRHi
+
+Target Cluster
+on Amazon OpenSearch Service ^idO17qxQ
+
+Middleware ^obMwP28y
+
+Cluster ^MrQCqzaS
+
+Backfill ^w7NFe2ol
+
+Metrics Source ^xUHSpl7U
+
+RFS on ECS Backfill Model ^QxHV7dcR
+
+Environment ^XWtNM3xv
+
+Environment
++ command
++ args ^5cio2Ikl
+
+Interfaces ^iablSv0Q
+
+Implementation ^MmC7YsUc
+
+Cloudwatch Metrics Model ^VX76m13O
+
+Environment ^lrTOHtQJ
+
+- get an empty one
+- get one based on YAML
+
+All middleware layer functions
+get passed an Environment, plus
+args ^z9HycCiZ
+
+DMS API ^CTJ9Hu4D
```

Comment:
Lets remove any references to a DMS API - REST/CLI should be enough

Reply:
Since the interface isn't locked in and its effectively built using the REST interface it doesn't really impact the architecture of the MA console, what do you think?

Reply:
Yeah, I've replaced this with just discussing the Web API and additional frontends